libsplinter/src/admin/service/shared.rs

```diff
@@ -146,6 +152,7 @@ pub struct AdminServiceShared {
     event_store: Box<dyn AdminServiceStore>,
     #[cfg(feature = "challenge-authorization")]
     public_keys: Vec<Vec<u8>>,
+    token_to_peer: HashMap<PeerAuthorizationToken, PeerNodePair>,
```

Comment:
When adding a peer_ref for a peer, the token_to_peer map is updated to include that peer so that we can map the token to our local peer authorization as well as the admin service for the PeerAuthorizationToken when handling connection/disconnection notifications. If all peer refs for that peer have been removed, we will also remove the entry in this map.